File: libs/grpc/clients/bots.ts
import { ChannelCredentials, createChannel, createClient } from 'nice-grpc';

import { CLIENT_OPTIONS, createClientAddr, waitReady } from './helper.js';
import { BotsClient, BotsDefinition } from '../bots/bots.client.js';
import { PORTS } from '../constants/constants.js';

export const createBots = async (env: string): Promise<BotsClient> => {
	const channel = createChannel(
		createClientAddr(env, 'bots', PORTS.BOTS_SERVER_PORT),
		ChannelCredentials.createInsecure(),
		CLIENT_OPTIONS,
	);

	await waitReady(channel);

	const client = createClient(BotsDefinition, channel);

	return client as any;
};

File: libs/grpc/clients/bots.go
package clients

import (
	"log"

	"github.com/twirapp/twir/libs/grpc/bots"
	"github.com/twirapp/twir/libs/grpc/constants"
	"google.golang.org/grpc"
)

func NewBots(env string) bots.BotsClient {
	serverAddress := createClientAddr(env, "bots", constants.BOTS_SERVER_PORT)

	conn, err := grpc.Dial(
		serverAddress,
		defaultClientsOptions...,
	)
	if err != nil {
		log.Fatalf("did not connect: %v", err)
	}
	c := bots.NewBotsClient(conn)
	return c
}
